src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1070,28 +1084,34 @@ internal static MethodSymbol TryFindBaseImplementationInInterface(NamedTypeSymbo
         private static Symbol FindMostSpecificImplementation(Symbol interfaceMember, NamedTypeSymbol implementingInterface)
         {
             HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-            var implementingMember = FindImplementationInInterface(interfaceMember, implementingInterface) ??
-                                     FindMostSpecificImplementation(interfaceMember, implementingInterface,
-                                                                    ref useSiteDiagnostics,
-                                                                    out var _, out var _);
-            return implementingMember;
+            MultiDictionary<Symbol, Symbol>.ValueSet implementingMember = FindImplementationInInterface(interfaceMember, implementingInterface);
+
+            switch (implementingMember.Count)
+            {
+                case 0:
+                    return FindMostSpecificImplementationInBases(interfaceMember, implementingInterface,
+                                                                 ref useSiteDiagnostics,
+                                                                 out var _, out var _);
+                case 1:
+                    return implementingMember.Single();
+                default:
+                    return null;
+            }
         }
 
         /// <summary>
         /// One implementation M1 is considered more specific than another implementation M2 
         /// if M1 is declared on interface T1, M2 is declared on interface T2, and 
         /// T1 contains T2 among its direct or indirect interfaces.
         /// </summary>
-        private static Symbol FindMostSpecificImplementation(Symbol interfaceMember,
-                                                             TypeSymbol implementingType,
-                                                             ref HashSet<DiagnosticInfo> useSiteDiagnostics,
-                                                             out Symbol conflictingImplementation1,
-                                                             out Symbol conflictingImplementation2)
+        private static Symbol FindMostSpecificImplementationInBases(
+            Symbol interfaceMember,
+            TypeSymbol implementingType,
+            ref HashSet<DiagnosticInfo> useSiteDiagnostics,
+            out Symbol conflictingImplementation1,
```

Comment:
I don't know if this was commented on in a previous PR, but why do we have two out params for conflicting implementations instead of a list or some other dynamically sizable thing? Why can there only ever be two conflicting implementations?

Reply:
> I don't know if this was commented on in a previous PR, but why do we have two out params for conflicting implementations instead of a list or some other dynamically sizable thing? Why can there only ever be two conflicting implementations?

Having two is enough to have a conflict, there is no benefit to track more because diagnostics reports just two.

---
In reply to: [262255212](https://github.com/dotnet/roslyn/pull/33778#discussion_r262255212) [](ancestors = 262255212)